gnuradio-runtime/include/gnuradio/buffer.h

```diff
@@ -135,7 +135,7 @@ namespace gr {
     std::multimap<uint64_t,tag_t>::iterator get_tags_lower_bound(uint64_t x) { return d_item_tags.lower_bound(x); }
     std::multimap<uint64_t,tag_t>::iterator get_tags_upper_bound(uint64_t x) { return d_item_tags.upper_bound(x); }
 
-    // -------------------------------------------------------------------------
+    virtual bool allocate_buffer();
```

Comment:
This was made public since the OOT blocks need to be able to access the (presumably subclassed) buffer allocation function during the block::allocate_output_buffer() call (at least this is how I ended up implementing things in my OpenCL test OOT module). I was unable to come up with a way for the block->buffer allocation to happen without this being public.